sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java

```diff
@@ -1,43 +1,82 @@
 package io.sentry.android.core;
 
+import androidx.annotation.NonNull;
 import io.sentry.IPerformanceContinuousCollector;
 import io.sentry.ISpan;
 import io.sentry.ITransaction;
 import io.sentry.NoOpSpan;
 import io.sentry.NoOpTransaction;
 import io.sentry.SentryDate;
 import io.sentry.SpanDataConvention;
-import io.sentry.SpanId;
 import io.sentry.android.core.internal.util.SentryFrameMetricsCollector;
 import io.sentry.protocol.MeasurementValue;
+import io.sentry.util.Objects;
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentSkipListSet;
 import java.util.concurrent.TimeUnit;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
 
 @ApiStatus.Internal
 public class SpanFrameMetricsCollector
     implements IPerformanceContinuousCollector,
         SentryFrameMetricsCollector.FrameMetricsCollectorListener {
 
-  private @NotNull final Object lock = new Object();
-  private @Nullable final SentryFrameMetricsCollector frameMetricsCollector;
-  private @Nullable volatile String listenerId;
-  private @NotNull final Map<SpanId, SentryFrameMetrics> metricsAtSpanStart;
+  // 30s span duration at 120fps = 3600 frames
+  private static final int MAX_FRAMES_COUNT = 3600;
```

Comment:
Please, don't make any assumption on frame numbers. IOS can, but Android is a different thing. There are devices with 144, 165, and even [240](https://www.gsmarena.com/sharp_aquos_zero_2-10259.php) fps at the moment. And I'm pretty sure it will increase over time 😅 
Also, Android doesn't call the callback on any frame drawn, but only when frames change. 
We can avoid checking frames count imho

Reply:
This is just an upper limit to ensure we don't run into OOMs when we have long running spans, as we collect every frame metric callback data.

Reply:
should we use the upper upper limit then? (30 * 240)?

Reply:
we can keep this number, as we will only receive slow frames anyway (normal frames are not recorded), but let's change the comment to explicit this is just a constraint to avoid OOMs